tests/test_sqlalchemy.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from typing import Optional
+
+import pytest
+from sqlalchemy import String
+from sqlalchemy.orm import DeclarativeBase, MappedAsDataclass
+from sqlalchemy.orm import Mapped
+from sqlalchemy.orm import mapped_column
+
+import serde
+from .common import (
+    opt_case,
+    opt_case_ids,
+    FormatFuncs,
+    format_dict,
+    format_tuple,
+    format_json,
+    format_msgpack,
+    format_yaml,
+    format_pickle,
+    format_toml,
+)
+
+log = logging.getLogger("test")
+
+serde.init(True)
+
+all_except_toml: FormatFuncs = (
+    format_dict + format_tuple + format_json + format_msgpack + format_yaml + format_pickle
+)
+
+
+@pytest.mark.parametrize("opt", opt_case, ids=opt_case_ids())
+@pytest.mark.parametrize("se,de", all_except_toml)
+def test_sqlalchemy_simple(se, de, opt):
+    log.info(f"Running test with se={se.__name__} de={de.__name__} opts={opt}")
+
+    class Base(MappedAsDataclass, DeclarativeBase):
+        pass
+
+    @serde.serde(**opt)
+    class User(Base):
+        __tablename__ = "user"
+
+        id: Mapped[int] = mapped_column(primary_key=True)
+        name: Mapped[str]
+        fullname: Mapped[str] = mapped_column(String(30))
+        nickname: Mapped[Optional[str]] = mapped_column(init=False)
+
+    user = User(1, "john", "John Doe")
+    assert user == de(User, se(user))
+
+
+@pytest.mark.parametrize("opt", opt_case, ids=opt_case_ids())
+@pytest.mark.parametrize("se,de", format_toml)
+def test_sqlalchemy_simple_toml(se, de, opt):
+    log.info(f"Running test with se={se.__name__} de={de.__name__} opts={opt}")
+
+    class Base(MappedAsDataclass, DeclarativeBase):
+        pass
+
+    @serde.serde(**opt)
+    class User(Base):
+        __tablename__ = "user"
+
+        id: Mapped[int] = mapped_column(primary_key=True)
+        name: Mapped[str]
+        fullname: Mapped[str] = mapped_column(String(30))
+
+    user = User(1, "john", "John Doe")
+    assert user == de(User, se(user))
```

Comment:
Can you run tests without sqlalchemy?

You can manually uninstall sqlalchemy and run tests here IMO
https://github.com/yukinarit/pyserde/blob/main/.github/workflows/test.yml#L32-L38

Reply:
Sure can do, but I have a unrelated doubt regarding https://github.com/yukinarit/pyserde/blob/main/.github/workflows/test.yml#L34
All tests are being run through `poetry run ...` while orjson is being installed (seemingly) outside the virtual enviroment with `pip install orjson`, it should be `poetry run pip install orjson` or do I miss something?

Reply:
Good catch. This line had been added two years ago by [this commit](https://github.com/yukinarit/pyserde/commit/2744675cd5af70426411d2e15aeb536e0c07c02b) but apparently pyserde was never tested with orjson :rofl: